src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocLeadingAsteriskAlignCheck.java

```diff
@@ -0,0 +1,196 @@
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.GlobalStatefulCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks that the
+ * <a href="https://docs.oracle.com/en/java/javase/14/docs/specs/javadoc/doc-comment-spec.html#leading-asterisks">
+ * leading asterisks</a> are aligned vertically under the second asterisk ( &#42; )
+ * of opening Javadoc tag. The alignment of closing Javadoc tag is also checked.
+ * If the line does not contain leading asterisk then that line will be ignored.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations if the
+ * Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">Tight-HTML Rules</a>.
+ * Type is {@code boolean}.
+ * Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.asterisk.indentation}
+ * </li>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.unclosedHtml}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 10.16.0
+ */
+@GlobalStatefulCheck
+public class JavadocLeadingAsteriskAlignCheck extends AbstractJavadocCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_WRONG_ALIGNMENT = "javadoc.asterisk.indentation";
+
+    /** Specifies the starting comment block of javadoc. */
+    private DetailNode parent;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.LEADING_ASTERISK,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    // get the starting block of javadoc
+    @Override
+    public void beginJavadocTree(DetailNode rootAst) {
+        parent = rootAst;
+    }
+
+    // process leading asterisks of the javadoc
+    @Override
+    public void visitJavadocToken(DetailNode ast) {
+        if (ast.getLineNumber() != parent.getLineNumber()) {
```

Comment:
Please help me to understand why we need this condition.

Reply:
It is used to check if the leading asterisk and the javadoc starting block are on the same line or not, if they're on same line then we don't have to check for the asterisk alignment.  Same goes for ending block of the comment, If the ending block is on the same line as starting block of comment, i.e it is signle line comment, then we don't need to check the alignment.

Examples:

```java
  /*************************************************
   *** @param str testing.....
   **********************************/
  public InputJavadocLeadingAsteriskAlignCorrect(String str) {}
```

```java
  /** javadoc for instance variable. */
  private int a;
```

@nrmancuso

Reply:
It is always helpful to show the AST:
```
➜  src cat Test.java 
public class Test {

    /************************************************
     **** @param str testing.....
     **********************************/
    public static class Inner {
    }

}

➜  src java -jar checkstyle-10.13.0-all.jar -J Test.java
COMPILATION_UNIT -> COMPILATION_UNIT [1:0]
`--CLASS_DEF -> CLASS_DEF [1:0]
    |--MODIFIERS -> MODIFIERS [1:0]
    |   `--LITERAL_PUBLIC -> public [1:0]
    |--LITERAL_CLASS -> class [1:7]
    |--IDENT -> Test [1:13]
    `--OBJBLOCK -> OBJBLOCK [1:18]
        |--LCURLY -> { [1:18]
        |--CLASS_DEF -> CLASS_DEF [6:4]
        |   |--MODIFIERS -> MODIFIERS [6:4]
        |   |   |--BLOCK_COMMENT_BEGIN -> /* [3:4]
        |   |   |   |--COMMENT_CONTENT -> ***********************************************\n     **** @param str testing.....\n     ********************************* [3:6]
        |   |   |   |   `--JAVADOC -> JAVADOC [3:7]
        |   |   |   |       |--LEADING_ASTERISK -> * [3:7]
        |   |   |   |       |--TEXT -> ********************************************* [3:8]
        |   |   |   |       |--NEWLINE -> \n [3:53]
        |   |   |   |       |--LEADING_ASTERISK ->      * [4:0]
        |   |   |   |       |--TEXT -> *** @param str testing..... [4:6]
        |   |   |   |       |--NEWLINE -> \n [4:33]
        |   |   |   |       |--LEADING_ASTERISK ->      * [5:0]
        |   |   |   |       |--TEXT -> ******************************** [5:6]
        |   |   |   |       `--EOF -> <EOF> [5:38]
        |   |   |   `--BLOCK_COMMENT_END -> */ [5:38]
        |   |   |--LITERAL_PUBLIC -> public [6:4]
        |   |   `--LITERAL_STATIC -> static [6:11]
        |   |--LITERAL_CLASS -> class [6:18]
        |   |--IDENT -> Inner [6:24]
        |   `--OBJBLOCK -> OBJBLOCK [6:30]
        |       |--LCURLY -> { [6:30]
        |       `--RCURLY -> } [7:4]
        `--RCURLY -> } [9:0]

```
Our Javadoc AST leaves something to be desired, it is strange to consider the asterisk at 3:7 "leading"

Reply:
> It is always helpful to show the AST:

will keep this in mind 🫡

> it is strange to consider the asterisk at 3:7 "leading"

I guess it is not strange because it is indicating the first `*` asterisk after the `/**` which should be considered as a leading asterisk after the declaration of the comment.